gpt4all-chat/src/xlsxtomd.cpp

```diff
@@ -33,31 +35,32 @@ static QString formatCellText(const QXlsx::Cell *cell)
     if (format.isDateTimeFormat()) {
         // Handle DateTime
         QDateTime dateTime = value.toDateTime();
-        cellText = dateTime.isValid() ? dateTime.toString("yyyy-MM-dd") : value.toString();
+        cellText = dateTime.isValid() ? dateTime.toString(QStringView(u"yyyy-MM-dd")) : value.toString();
     } else {
         cellText = value.toString();
     }
 
     if (cellText.isEmpty())
         return QString();
 
-    // Apply Markdown and HTML formatting based on font styles
-    QString formattedText = cellText;
-
-    if (format.fontBold() && format.fontItalic())
-        formattedText = "***" + formattedText + "***";
-    else if (format.fontBold())
-        formattedText = "**" + formattedText + "**";
-    else if (format.fontItalic())
-        formattedText = "*" + formattedText + "*";
-
+    // Escape special characters
+    static QRegularExpression special(uR"([\\`*_{}[\]()#+-.!])"_s);
+    cellText.replace(special, uR"(\\1)"_s);
```

Comment:
The escaping is worse. I am getting this on the walt disney xlsx example for produced markdown:

`| | | | | | | |
|---|---|---|---|---|---|---|
|~~***_Walt Disney Co\\1_***~~| | | | | | |
|~~***_Consolidated Income Statement_***~~| | | | | | |
| | | | | | | |
|US$ in millions| | | | | | |
|~~***_12 months ended:_***~~|~~***_45199_***~~|~~***_44835_***~~|~~***_44471_***~~|~~***_44107_***~~|~~***_43736_***~~|~~***_43372_***~~|
|Services|79562\\10|74200\\10|61768\\10|59265\\10|60542\\10|50869\\10|
|Products|9336\\10|8522\\10|5650\\10|6123\\10|9028\\10|8565\\10|
|~~***_Revenues_***~~|~~***_88898\\10_***~~|~~***_82722\\10_***~~|~~***_67418\\10_***~~|~~***_65388\\10_***~~|~~***_69570\\10_***~~|~~***_59434\\10_***~~|
|Cost of services\\1 exclusive of depreciation and amortization|\\153139\\10|\\148962\\10|\\141129\\10|\\139406\\10|\\136450\\10|\\127528\\10|`

Which looks like this:

![image](https://github.com/user-attachments/assets/859d1e4a-110b-4f51-bfec-12d485bf4622)

Reply:
Notice the strikeout as well...

Reply:
There was a mistake in my regex and a bug in QXlsx. I have fixed both of them.